backend/common/src/main/java/ai/verta/modeldb/common/futures/FutureUtil.java

```diff
@@ -0,0 +1,81 @@
+package ai.verta.modeldb.common.futures;
+
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import io.grpc.Context;
+import io.opentracing.Scope;
+import io.opentracing.util.GlobalTracer;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ForkJoinPool;
+import org.springframework.lang.NonNull;
+
+public abstract class FutureUtil {
```

Comment:
this has no abstract methods and it makes no sense for things to inherit this vs just use directly. inheriting seems to make the indirection more complex and the code less obvious. is there a reason this isn't just a final concrete class?

Reply:
Fixed it with final class.